### PR TITLE
Small fixes to NTLM code

### PR DESCRIPTION
**Pull request by elric1**

### CODE REVIEW

File: lib/gssapi/ntlm/init_sec_context.c [hunk 1]
```diff
@@ -381,6 +381,7 @@ _gss_ntlm_init_sec_context
 		if (RAND_bytes(nonce, sizeof(nonce)) != 1) {
 		    _gss_ntlm_delete_sec_context(minor_status,
 						 context_handle, NULL);
+		    heim_ntlm_free_type2(&type2);
```
Comment (jaltman) [inline lib/gssapi/ntlm/init_sec_context.c:384]: I think these are all the leak points for type2 but this function needs refactoring to make it reviewable.

File: lib/ntlm/ntlm.c [hunk 2]
```diff
@@ -718,6 +718,9 @@ heim_ntlm_decode_type1(const struct ntlm_buf *buf, struct ntlm_type1 *data)
 /**
  * Encodes an ntlm_type1 message.
  *
+ *     [MS-NLMP] 2.2.1.1 NEGOTIATE_MESSAGE
+ *     https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-nlmp/
+ *
```
Comment (jaltman) [inline lib/ntlm/ntlm.c:723]: I cannot reasonably review this change nor do I have a way of testing it.

File: lib/gssapi/ntlm/accept_sec_context.c [hunk 3]
```diff
@@ -177,6 +177,7 @@ _gss_ntlm_accept_sec_context
 	}
 	memcpy(output_token->value, out.data, out.length);
 	output_token->length = out.length;
+	heim_ntlm_free_buf(&out);
```
[the remark this answers is not in the document]
Comment (elric1) [reply lib/gssapi/ntlm/accept_sec_context.c:180]: The former, yes. The latter, no, it leaves out undefined on failure. I'll fix that, too.

File: lib/gssapi/ntlm/delete_sec_context.c [hunk 4]
```diff
@@ -49,6 +51,8 @@ OM_uint32 GSSAPI_CALLCONV _gss_ntlm_delete_sec_context
 	    (*ctx->server->nsi_destroy)(minor_status, ctx->ictx);
 
 	_gss_ntlm_release_cred(NULL, &cred);
+	krb5_data_free(&ctx->sessionkey);
+	gss_release_buffer(&min, &ctx->pac);
 
 	memset(ctx, 0, sizeof(*ctx));
```
[the remark this answers is not in the document]
Comment (elric1) [reply lib/gssapi/ntlm/delete_sec_context.c:59]: Okay, added.

**Comment by jaltman** [comment]
The memleak changes have been merged 
8dcd05ed4  _gss_ntlm_init_sec_context() mem leaks
e87fca809 _gss_ntlm_delete_sec_context() mem leaks
8526b4c62 fix memory leak near NTLM type2 response

File: lib/gssapi/ntlm/init_sec_context.c [hunk 5]
```diff
@@ -365,7 +365,7 @@ _gss_ntlm_init_sec_context
 
 	type3.username = ctx->client->username;
 	type3.flags = type2.flags;
-	type3.targetname = type2.targetname;
+	type3.targetname = name->domain;
```
Comment (nicowilliams) [inline lib/gssapi/ntlm/init_sec_context.c:368]: Above this are some XXX comments:

```
        /* XXX check that type2.targetinfo matches `target_name´ */
        /* XXX check verify targetinfo buffer */
```

Also, apparently the `targetname` of type 3 can be either a domain or a server name, yes?  How would we know which we should use here?

File: lib/ntlm/ntlm.c (hunk 2 above)
Comment (nicowilliams) [reply lib/ntlm/ntlm.c:723]: The linked spec does say this (`SHOULD`).

**Pull request closed without merging**